Automattic/woocommerce-payments
# Improve the wording and style of the "Active Disputes" task list item on the Payments → Overview screen to highlight urgency

* Fixes #6370
* Fixes #6404
* Fixes #6403
* Fixes #6485

#### Changes proposed in this Pull Request

This PR updates the title, subtitle and colours and action button label for the Payments → Overview `Things to do` active disputes task list item. It also makes the task dismissible and ensures the task will appear for stores who have previously dismissed this task.

![payment-overview](https://github.com/Automattic/woocommerce-payments/assets/3147296/986f469e-0a5b-47b9-831b-8b8262569cc4)

**Single/multiple disputes with > 7 days until due**
* **Visibility**: Not visible

**Single dispute with < 7 days until due**
* **Visibility/colour**: 🟡 `Yellow`
* **Title**: `Respond to a dispute for $123.00`
* **Subtitle**: `By Jun 30, 2023 – 1 week left to respond`
* **Action label**: `Respond now`
* **Action redirect**: Single dispute details page

**Single dispute with < 24 hours until due**
* **Visibility/colour**: 🔴 `Red`
* **Title**: `Respond to a dispute for $123.00 – Last day`
* **Subtitle**: `Respond today by 11:59 PM`
* **Action label**: `Respond now`
* **Action redirect**: Single dispute details page

**Multiple disputes with at least one due within 7 days**
* **Visibility/colour**: 🟡 `Yellow`
* **Title**: `Respond to N active disputes for $123.00, €123.00`
* **Subtitle**: `Last week to respond to N of the disputes`
* **Action label**: `See disputes`
* **Action redirect**: Active disputes list screen

**Multiple disputes with at least one due within 24 hours**
* **Visibility/colour**: 🔴 `Red`
* **Title**: `Respond to N active disputes for $123.00, €123.00`
* **Subtitle**: `Final day to respond to N of the disputes`
* **Action label**: `See disputes`
* **Action redirect**: Active disputes list screen

**Tracks event**

The unregistered tracks event `wcpay_overview_task` has been renamed `wcpay_overview_task_click` to meet tracks event naming guidelines. This event was only in use for the task `dispute_resolution_task`. It now accepts the event property `task` and `active_dispute_count`. It will be registered once this PR has been merged: see issue #6541.

**Design/component library tradeoffs**

The task list items use the `TaskItem` component from `@woocommerce/experimental`.

This raises some issues/limitations:

1. The ❗ icon does not appear if the `TaskItem` `level` is `2` (yellow).
1. There is a tooltip that displays when hovering the icon/checkbox:
	* level 1 (red): `This task is required to keep your store running`
	* level 2 (yellow): `This task is required to set up your extension`

💡 I've solved this by using level 1 for both red/yellow and overriding colour via CSS.

---

### Testing instructions





* Tip: To create a disputed transaction, purchase a product with the Stripe test card `4000000000000259`.
* View the Payments → Overview screen with the following configuration of disputes and confirm it matches the requirements above:
	* 0 active disputes (task not visible)
	* 1+ active disputes due more than 7 days in the future (task not visible)
	* 1 active dispute due within 7 days
	* 1 active dispute due within 24h
	* 2+ active disputes of multiple currencies due within 7 days
	* 2+ active disputes of multiple currencies due within 24h
* Check that the task list looks correct with a non-dispute task item enabled alongside the dispute task item.
	* Mark these tasks as visible by returning `true` for `showUpdateDetailsTask` and `wpcomReconnectUrl` in [`WC_Payments_Admin::get_js_settings()`](https://github.com/Automattic/woocommerce-payments/blob/5a3909d4823d3794f53f73fed5d468ca905ca656/includes/admin/class-wc-payments-admin.php#L763-L764).

* Check that the currency amounts look correct after modifying the WC country:
	* Navigate to WooCommerce > Settings > General.
	* Change the Country / State to a region with non-decimal (Japan, Yemen) or three-decimal (Bahrain, Kuwait) display.

**Mocking dispute due_by when testing**

To mock dispute `due_by` dates testing purposes, add the following logic and adjust the 'days' value: https://github.com/Automattic/woocommerce-payments/blob/56dd736e4b2d335ffcdd4e4a6b676d7df0861c78/client/overview/index.js#L75
```diff
-		activeDisputes,
+		activeDisputes: activeDisputes.map( ( dispute ) => {
+			const days = 6;
+			dispute.due_by = new Date(
+				Date.now() + days * 24 * 60 * 60 * 1000
+			).toISOString();
+			return dispute;
+		} ),
```

**Screenshot with multiple currency disputes, a non-decimal country (Japan) and other tasks**
![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/2092668a-e5cf-4b5d-8dfd-5b0c4f5ab8fc)

-------------------


**Post merge**
